### PR TITLE
Fix Windows-hosted builds and toolchain injection

### PR DESCRIPTION
**Pull request by Th3T3chn0G1t**
Fixed hosting of builds under typical Windows shells and added a more robust way to inject toolchain files from a secondary project

### CODE REVIEW

File: installer/hyper_install.c [hunk 1]
```diff
@@ -174,7 +174,7 @@ void ensure_no_gpt(FILE *img)
 int main(int argc, char **argv)
 {
     struct mbr_partition_entry mbr_partitions[MBR_PARTITION_COUNT];
-    bool is_iso;
+    bool is_iso = argc == 3 && !strcmp(argv[2], "iso");
```
Comment (d-tatianin) [inline installer/hyper_install.c:177]: Maybe something like `--force-iso` would be better?

File: installer/hyper_install.c [hunk 2]
```diff
@@ -184,7 +184,7 @@ int main(int argc, char **argv)
     img = safe_open(argv[1], "r+b");
 
     read_mbr_partition_list(img, mbr_partitions);
-    is_iso = is_iso_disk(img);
+    is_iso = is_iso || is_iso_disk(img);
```
Comment (d-tatianin) [inline installer/hyper_install.c:187]: `is_iso |= is_iso_disk(img)`

File: CMakeLists.txt [hunk 3]
```diff
@@ -33,19 +33,20 @@ if (HYPER_TOOLCHAIN)
 endif ()
 
 if (NOT HYPER_TOOLCHAIN)
-    set(HYPER_TOOLCHAIN "gcc")
-elseif (NOT HYPER_TOOLCHAIN STREQUAL "gcc" AND
-        NOT HYPER_TOOLCHAIN STREQUAL "clang")
-    message(FATAL_ERROR "Unknown toolchain '${HYPER_TOOLCHAIN}'")
-endif ()
-
-if ($ENV{CLION_IDE})
```
Comment (d-tatianin) [inline CMakeLists.txt:42]: I don't necessarily disagree with moving clion detection under that if statement, but that seems like it belongs in a separate commit?

Something like: `don't attempt to detect clion if a toolchain is explicitly specified` (or better wording 🙂 )

**Review by d-tatianin — CHANGES_REQUESTED** [review]
Thanks a lot, these do look like very useful changes!

Just a few minors nits and this is good for merging I think.

File: installer/hyper_install.c (hunk 2 above)
Comment (Th3T3chn0G1t) [reply installer/hyper_install.c]: Change from my own testing I forgot to remove -- fixed

File: CMakeLists.txt (hunk 3 above)
Comment (Th3T3chn0G1t) [reply CMakeLists.txt:42]: Moved to "build: disable CLion workaround if toolchain is explicitly specified" cbab01b

**Comment by d-tatianin** [comment]
LGTM!
